File: zadanie8-1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

#define MAX_NODES 200
#define INF 1000000000
#define min(a, b) ((a) < (b) ? (a) : (b))

int n; // number of nodes
int c[MAX_NODES][MAX_NODES]; // capacity matrix
int f[MAX_NODES][MAX_NODES]; // flow matrix
int d[MAX_NODES]; // distance array
int p[MAX_NODES]; // parent array

int kde_ideme;
int running_sum = 0;

// Breadth-first search to find the shortest augmenting path
bool bfs(int s, int t) {
    memset(d, -1, sizeof(d));
    d[s] = 0;
    p[s] = -1;
    int q[MAX_NODES];
    int front = 0, rear = 0;
    q[rear++] = s;
    while (front < rear) {
        int u = q[front++];
        for (int v = 0; v < n; v++) {
            if (d[v] == -1 && f[u][v] < c[u][v]) {
                d[v] = d[u] + 1;
                p[v] = u;
                q[rear++] = v;
            }
        }
    }
    return d[t] != -1;
}

// Augment flow along the shortest augmenting path
int dfs(int u, int t, int flow) {
    if (u == t) return flow;
    for (int v = 0; v < n; v++) {
        if (f[u][v] < c[u][v] && d[v] == d[u] + 1) {
            int df = dfs(v, t, min(flow, c[u][v] - f[u][v]));
            if (df > 0) {
                f[u][v] += df;
                f[v][u] -= df;
                return df;
            }
        }
    }
    return 0;
}

// Compute the maximum flow from s to t using the Edmonds-Karp algorithm
int edmonds_karp(int s, int t) {
    int max_flow = 0;
    while (bfs(s, t)) {
        int flow;
        do {
            flow = dfs(s, t, INF);
            max_flow += flow;
        } while (flow > 0);
    }
    return max_flow;
}

int main() {
    // Read input and initialize capacity matrix
    int a;
    int b;
    while (scanf("%d",&a) != EOF)
    {

        scanf("%d",&b);
        memset(c, 0, sizeof(c));
        memset(f, 0, sizeof(f));


        int nedavam_to[100];
        int nedavam_tot[100];

        int vstup;
        int pocitam = 0;


        n = a+b;
        kde_ideme = n+1;

        for (int i = 0; i < n; i++)
        {
            scanf("%d", &vstup);
            if (i < a)
            {
                running_sum += vstup;
                nedavam_to[i] = vstup;
            }
            else
            {                
                nedavam_tot[i-a] = vstup;
            }
        }


        int k = 0;
        for (int i = 0; i < n; i++)
        {
            if (i < a)
            {
                c[0][i+1] += nedavam_to[i];
                for (int j = 0; j < b; j++)
                {
                    c[i+1][a+j+1] += 1;
                }
                
            }
            else
            {
                c[i+1][a+b+1] += nedavam_tot[i-a];
            }
        }
        for (int i = 0; i < 11; i++)
        {
            for (int j = 0; j < 11; j++)
            {
                printf("%d",c[i][j]);
            }
            printf("\n");
        }
    
        printf("%d\n", edmonds_karp(0, n-1));
    }

    // Compute and print the maximum flow from node 0 to node n-1
    
    return 0;
}

File: zadanie8-1.py

deti = [4, 5, 3, 5] 
stoly =  [3, 5, 2, 6, 4]

for i in range(len(stoly)):
    if stoly[i] > len(deti):
        stoly[i] = len(deti)

from collections import deque

class Edge:
    def __init__(self, u, v, w):
        self.source = u
        self.target = v
        self.capacity = w

class Graph:
    def __init__(self):
        self.adj = {}
        self.flow = {}

    def add_edge(self, u, v, w):
        if u not in self.adj:
            self.adj[u] = []
        if v not in self.adj:
            self.adj[v] = []
        self.adj[u].append(Edge(u, v, w))
        self.adj[v].append(Edge(v, u, 0))
        self.flow[(u, v)] = 0
        self.flow[(v, u)] = 0

    def bfs(self, s, t, parent):
        visited = set()
        queue = deque()
        queue.append(s)
        visited.add(s)

        while queue:
            u = queue.popleft()
            for edge in self.adj[u]:
                v = edge.target
                if v not in visited and self.flow[(u, v)] < edge.capacity:
                    parent[v] = (u, edge)
                    if v == t:
                        return True
                    queue.append(v)
                    visited.add(v)

        return False

    def ford_fulkerson(self, source, sink):
        parent = {}
        max_flow = 0

        while self.bfs(source, sink, parent):
            path_flow = float('inf')
            s = sink
            while s != source:
                u, edge = parent[s]
                path_flow = min(path_flow, edge.capacity - self.flow[(u, s)])
                s = u

            max_flow += path_flow

            s = sink
            while s != source:
                u, edge = parent[s]
                self.flow[(u, s)] += path_flow
                self.flow[(s, u)] -= path_flow
                s = u

        return max_flow
    


# create a graph
g = Graph()

# # add edges to the graph
j = 0


for i in deti:
    g.add_edge('S', str(j),i)
    k = len(deti)
    for l in stoly:
        g.add_edge(str(j), str(k),1)
        g.add_edge(str(k), "T", l)
        k += 1
    j += 1


# get the maximum flow from node 'S' to node 'T'
max_flow = g.ford_fulkerson('S', 'T')

if max_flow < sum(deti):
    print(0)
else:
    print(1)